File: Solutions/Python/grid.py
# Getting necessary information
rows = int(input("Enter the number of rows: "))
columns = int(input("Enter the number of columns: "))
# Initializing matrix with desired dimensions
grid = [[0 for j in range(columns)] for i in range(rows)]
# Assigning every point the value described by solution
for i in range(rows):
    for j in range(columns):
        if j == 0 or i == 0:
            grid[i][j] = 1
        else:
            grid[i][j] = grid[i - 1][j] + grid[i][j - 1]
# Printing last element of matrix
print("The different paths are:", grid[-1][-1])

File: Solutions/C/grid.c
#include <stdio.h>

/* Data variables */
int columns, rows;

int main() {

    /* Getting data */ 
    printf("\nEnter the number of rows: ");
    scanf("%d", &rows);
    printf("Enter the number of columns: ");
    scanf("%d", &columns);

    /* Creating the matrix(grid) */
    unsigned long long square[rows][columns];

    /* Executing the algorithm described in solution */
    for (int i = 0; i < rows; ++i) {
        for (int j = 0; j < columns; ++j) {
            if (j == 0 || i == 0) {
                square[i][j] = 1ULL;
            } 
            else {
                square[i][j] = square[i - 1][j] + square[i][j - 1];
            }
        }
    }

    /* Printing the result */
    printf("\nThe different paths are: %lld", square[--rows][--columns]);
    return 0;
}